examples/browser/test/pages/main-page.ts

```diff
@@ -57,15 +63,102 @@ export class MainPage {
     }
+
+    /* Click on OPEN in the context menu to open the file in the workspace */
+    public isContextMenuOpenAvailable(filePositionNumber: Number): Boolean {
+        browser.waitForVisible(`#files div.theia-TreeNode.theia-FileStatNode:nth-child(${filePositionNumber}`);
+        browser.rightClick(`#files div.theia-TreeNode.theia-FileStatNode:nth-child(${filePositionNumber})`, 0, 0);
+        browser.waitForExist('div.p-Widget.p-Menu ul.p-Menu-content');
+        browser.click('ul li.p-Menu-item:nth-child(1)');
+        return browser.isExisting('ul.p-TabBar-content');
+    }
+
+    /* Click on OPEN WITH in the context menu to open the file using Code Editor */
+    public isContextMenuOpenWithAvailable(filePositionNumber: Number): Boolean {
+        browser.waitForVisible(`#files div.theia-TreeNode.theia-FileStatNode:nth-child(${filePositionNumber}`);
+        browser.rightClick(`#files div.theia-TreeNode.theia-FileStatNode:nth-child(${filePositionNumber})`, 0, 0);
+        browser.waitForExist('div.p-Widget.p-Menu ul.p-Menu-content');
+        browser.click('ul li.p-Menu-item:nth-child(2)')
+        browser.waitForExist('div.p-Widget.p-Menu ul.p-Menu-content');
+        browser.click('ul li.p-Menu-item:nth-child(1)');
+        return browser.isExisting('ul.p-TabBar-content');
+    }
+
+    /* Click on NEW FILE in the context menu to create a new file in the workspace */
+    public isContextMenuNewFileAvailable(): Boolean {
+        var result = false;
+        browser.waitForVisible('#files div.theia-TreeNode.theia-FileStatNode');
+        browser.rightClick('#files div.theia-TreeNode.theia-FileStatNode', 0, 0);
+        browser.waitForExist('div.p-Widget.p-Menu ul.p-Menu-content');
+        browser.click('ul li.p-Menu-item:nth-child(10)');
+        browser.click('button.dialogButton').pause(300);
+        var folder_elements = browser.getText('div.theia-TreeNodeCaption').toString();
+        var elements = folder_elements.split(",", 100);
+        elements.forEach(element => {
+            if (element === 'Untitled.txt') {
+                result = true;
+            }
+        });
+        return result;
+    }
+
+    /* Click on NEW FOLDER in the context menu to create a new folder */
+    public isContextMenuNewFolderAvailable(): Boolean {
+        var result = false;
+        browser.rightClick('#theia-left-stack');
+        browser.waitForExist('div.p-Widget.p-Menu ul.p-Menu-content');
+        browser.click('ul li.p-Menu-item:nth-child(11)');
+        browser.waitForExist('div.dialogContent');
+        browser.click('button.dialogButton').pause(300);
+        var folder_elements = browser.getText('div.theia-TreeNodeCaption').toString();
+        var elements = folder_elements.split(",", 100);
+        elements.forEach(element => {
+            if (element === 'Untitled') {
+                console.log('**************************Untitled exists')
+                result = true;
+            }
+        });
+        return result;
```

Comment:
Please look into [`some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some). You could get rid of a few lines of code. Including the leftover `console.log`.

```typescript
const elements = ['a.txt', 'b.txt', 'Untitled.txt'];
return elements.some(e => e === 'Untitled.txt');
```